xcm-support/src/currency_adapter.rs

```diff
@@ -46,35 +64,48 @@ pub struct MultiCurrencyAdapter<MultiCurrency, Matcher, AccountIdConverter, Acco
 
 impl<
 		MultiCurrency: orml_traits::MultiCurrency<AccountId, CurrencyId = CurrencyId>,
+		UnknownAsset: UnknownAssetT,
 		Matcher: MatchesFungible<MultiCurrency::Balance>,
 		AccountIdConverter: LocationConversion<AccountId>,
 		AccountId: sp_std::fmt::Debug,
 		CurrencyIdConverter: CurrencyIdConversion<CurrencyId>,
 		CurrencyId: FullCodec + Eq + PartialEq + Copy + MaybeSerializeDeserialize + Debug,
 	> TransactAsset
-	for MultiCurrencyAdapter<MultiCurrency, Matcher, AccountIdConverter, AccountId, CurrencyIdConverter, CurrencyId>
+	for MultiCurrencyAdapter<
+		MultiCurrency,
+		UnknownAsset,
+		Matcher,
+		AccountIdConverter,
+		AccountId,
+		CurrencyIdConverter,
+		CurrencyId,
+	>
 {
 	fn deposit_asset(asset: &MultiAsset, location: &MultiLocation) -> Result {
-		let who = AccountIdConverter::from_location(location)
-			.ok_or_else(|| XcmError::from(Error::AccountIdConversionFailed))?;
-		let currency_id =
-			CurrencyIdConverter::from_asset(asset).ok_or_else(|| XcmError::from(Error::CurrencyIdConversionFailed))?;
-		let amount: MultiCurrency::Balance = Matcher::matches_fungible(&asset)
-			.ok_or_else(|| XcmError::from(Error::AssetNotFound))?
-			.saturated_into();
-		MultiCurrency::deposit(currency_id, &who, amount).map_err(|e| XcmError::FailedToTransactAsset(e.into()))?;
-		Ok(())
+		UnknownAsset::deposit(asset, location).or_else(|_| {
```

Comment:
should try known first and then try unknown

Reply:
Aha yes my bad. I thought it the wrong way (in order to make it return known asset deposit error). Will update.